### PR TITLE
Make `Html::resume_text()` result safe

### PR DESCRIPTION
**Pull request by cedric-anne**
## Checklist before requesting a review

- [x] I have read the CONTRIBUTING document.
- [x] I have performed a self-review of my code.
- [x] I have added tests that prove my fix is effective or that my feature works.

## Description

`Html::resume_text()` purpose is to truncate a text to be displayed in an HTML page. I propose to always encode special chars contained in the text to make it easier to use.

### CODE REVIEW

File: src/Html.php [hunk 1]
```diff
@@ -210,11 +211,13 @@ public static function cleanParametersURL($url)
      **/
     public static function resume_text($string, $length = 255)
     {
-        if (Toolbox::strlen($string) > $length) {
-            $string = Toolbox::substr($string, 0, $length) . "&nbsp;(...)";
+        $append = '';
+        if (mb_strlen($string, 'UTF-8')) {
```
Comment (AdrienClairembault) [inline src/Html.php:215]:
```suggestion
        if (mb_strlen($string, 'UTF-8') > $length) {
```

I don't know if you removed this check on purpose but without it any non empty string will be appended `&nbsp;(...)`.

File: src/Html.php [hunk 2]
```diff
@@ -210,11 +211,13 @@ public static function cleanParametersURL($url)
      **/
     public static function resume_text($string, $length = 255)
     {
-        if (Toolbox::strlen($string) > $length) {
-            $string = Toolbox::substr($string, 0, $length) . "&nbsp;(...)";
+        $append = '';
+        if (mb_strlen($string, 'UTF-8') > $length) {
```
Comment (trasher) [inline src/Html.php:215]: Any reason not to use `Toolbox::strlen()` sinc eit does exactly the same?

**Review by trasher — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

File: src/Html.php (hunk 2 above)
Comment (cedric-anne) [reply src/Html.php:215]: I am not a big fan of methods that exists just to not have to write a second parameter.

Comment (trasher) [reply src/Html.php:215]: OK. So maybe it should be deprecated and replaced then. But out of current scope obviously.

**Pull request merged**